From e3b957d0a10a5891cab77f508a0bdac6c3440f06 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 30 Apr 2017 17:44:42 +0300
Subject: [PATCH 0677/1200] arm64: mvebu: armada-8040-mcbin-db: enable
 networking in device tree

Enabled interfaces:
	- 10G SFI CP0, 10G SFI CP1, 2.5G SGMII CP1, 1G SGMII CP1;
1G SGMII CP1 is connected to CP0 mdio;
10G SFI CP0 and 10G SFI CP1 are connected to CP0 xmdio.

Change-Id: I79cae6dc465079d1ec4820aefbfcd15d9965a8f6
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/55399
Reviewed-by: Igal Liberman <igall@marvell.com>
Tested-by: Igal Liberman <igall@marvell.com>
---
 arch/arm/dts/armada-8040-mcbin.dts | 52 ++++++++++++++++++++++++++++++
 1 file changed, 52 insertions(+)

diff --git a/arch/arm/dts/armada-8040-mcbin.dts b/arch/arm/dts/armada-8040-mcbin.dts
index 4895413131..640da33f3e 100644
--- a/arch/arm/dts/armada-8040-mcbin.dts
+++ b/arch/arm/dts/armada-8040-mcbin.dts
@@ -186,10 +186,34 @@
 
 &cp0_mdio {
 	status = "okay";
+	ge_phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
 };
 
 &cp0_xmdio {
 	status = "okay";
+	sfi_phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+
+	sfi_phy8: ethernet-phy@8 {
+		reg = <8>;
+	};
+};
+
+&cp0_ethernet {
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+/* 10G SFI/copper */
+&cp0_eth0 {
+	status = "okay";
+	phy-mode = "sfi";
+	phy = <&sfi_phy0>;
+	phy-reset-gpios = <&cp1_gpio0 31 GPIO_ACTIVE_HIGH>;
+	marvell,sfp-tx-disable-gpio = <&cp1_gpio0 29 GPIO_ACTIVE_HIGH>;
 };
 
 &cp1_sata0 {
@@ -298,3 +322,31 @@
 		phy-speed = <COMPHY_SPEED_3_125G>;
 	};
 };
+
+&cp1_ethernet {
+	status = "okay";
+};
+
+/* 10G SFI/copper */
+&cp1_eth0 {
+	status = "okay";
+	phy-mode = "sfi";
+	phy = <&sfi_phy8>;
+	phy-reset-gpios = <&cp1_gpio0 9 GPIO_ACTIVE_HIGH>;
+	marvell,sfp-tx-disable-gpio = <&cp1_gpio0 10 GPIO_ACTIVE_HIGH>;
+};
+
+/* 1G SFI */
+&cp1_eth1 {
+	status = "okay";
+	phy-mode = "sgmii";
+	phy = <&ge_phy0>;
+};
+
+/* 2.5G copper */
+&cp1_eth2 {
+	status = "okay";
+	phy-mode = "sgmii";
+	phy-speed = <2500>;
+	marvell,sfp-tx-disable-gpio = <&cp1_gpio0 24 GPIO_ACTIVE_HIGH>;
+};
-- 
2.22.0

